### PR TITLE
Feat/umi integration

### PR DESCRIPTION
**Pull request by ashwini06**
## Background:
So far, the developed UMI workflow was run independently with balsamic cli with the option `--analysis-type umi` (#359 )
But the main goal is to run UMI workflow as an internal part of the main balsamic workflow for `tumor-only` and `targeted panel` cases. 

### This PR:
The option  ```--umiworkflow/--no-umi workflow``` is added to the config file, which commands the main balsamic workflow to execute umi workflow for `tumor-only `cases and `panel` data (default option). The output results `vcf` and `vep` folders consist of extra files with `*.umi*` tag

**Testing:** 
**case1:** Run balsamic and umi workflow for` tumoronly` and `panel` data 
```  
step1:
balsamic config case \
    -t ${_tumor_fastq} \
    ${_normal_option} \
    --case-id ${_case_name} \
    --analysis-dir ${_analysis_dir} \
    -r ${_reference} \
    ${_panel_option} \
    --singularity ${_singularity} \
    --background-variants ${_bg_variants} \
    --umiworkflow

step 2:
balsamic run analysis -s ${_analysis_config} -c ${_cluster_config} --account development ${_run_analysis}
```
tumoronly_panel_withumi_n2

**case2: ** Run **only balsamic** for` tumoronly` and `panel` data 

```  
step1:
balsamic config case \
    -t ${_tumor_fastq} \
    ${_normal_option} \
    --case-id ${_case_name} \
    --analysis-dir ${_analysis_dir} \
    -r ${_reference} \
    ${_panel_option} \
    --singularity ${_singularity} \
    --no-umiworkflow

step 2:
balsamic run analysis -s ${_analysis_config} -c ${_cluster_config} --account development ${_run_analysis}
```

tumoonly_panel_noumi_n2

VEP results folder with `*.umi*` vcf files
vep_results

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by ashwini06** [comment]
@hassanfa : This PR is not yet complete, there needs a discussion on this implementation. Maybe, we can go together next week on this :)

**Comment by hassanfa** [comment]
> @hassanfa : This PR is not yet complete, there needs a discussion on this implementation. Maybe, we can go together next week on this :)

It looks quite promising though. Tests passed and all. It just needs some ironing out.

File: BALSAMIC/utils/constants.py [hunk 1]
```diff
@@ -39,10 +39,22 @@
 MUTATION_CLASS = ["somatic", "germline"]
 MUTATION_TYPE = ["SNV", "SV", "CNV"]
 ANALYSIS_TYPES = ["paired", "single", "umi", "qc"]
-WORKFLOW_SOLUTION = ["BALSAMIC", "Sentieon", "DRAGEN"]
+WORKFLOW_SOLUTION = ["BALSAMIC", "Sentieon", "DRAGEN", "Sentieon_umi"]
```
Comment (hassanfa) [inline BALSAMIC/utils/constants.py:42]: 👍🏽

File: BALSAMIC/workflows/UMIworkflow.smk [hunk 2]
```diff
@@ -65,22 +63,20 @@ paramsumi = UMIworkflowConfig.parse_obj(umiworkflow_params)
 # Define wildcards
 SAMPLES = config["samples"]
 CASE_NAME = config["analysis"]["case_id"]
-VAR_CALLER = ["TNscope.umi","vardict.umi"]
+VAR_CALLER = ["TNscope","vardict"]
 ALL_STEPS = ["consensusaligned","consensusfiltered", "umialign"]
 FILTERED_STEPS = ["consensusaligned","consensusfiltered"]
-NEW_CASE_NAME = expand("{case_nm}.{step}", case_nm = CASE_NAME, step=FILTERED_STEPS)
+EXTN_NAME = expand("{var_caller}_{step}_umi", var_caller = ["TNscope","vardict"], step=FILTERED_STEPS)
```
Comment (hassanfa) [inline BALSAMIC/workflows/UMIworkflow.smk:69]: what does EXTN_NAME stand for?

**Review by hassanfa — APPROVED** [review]
Just a very minor comment.

**Pull request merged**